hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -136,6 +150,29 @@ private void rescheduleSelfWithBackoff() {
         }
     }
 
+    private FrameReadTaskSourceProvider createTaskSourceProvider(MeshSourceAggregator sourceAggregator) {
+        Query query = sourceAggregator.query;
+        int totalTasks = sourceAggregator.totalTasks;
+        int maxSimul = getMaxSimul(query.getParameter("maxSimul"), totalTasks);
+        if (maxSimul == totalTasks) {
+            return new DefaultFrameReadTaskSourceProvider(sourceAggregator);
+        } else {
+            return new MaxSimulFrameReadTaskSourceProvider(sourceAggregator, maxSimul);
+        }
+    }
+
+    private int getMaxSimul(@Nullable String maxSimulQueryParam, int totalTasks) {
+        if (maxSimulQueryParam == null) {
+            return totalTasks;
+        }
+        try {
```

Comment:
those cases don't seem terribly convincing, but in any event, using exceptions for control flow is usually not great anyway. If the intention here is to explicitly allow for malformed inputs, might as well use `Ints.tryParse`.